autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingActivityRepository.java

```diff
@@ -18,32 +19,38 @@
 
 @EntityType(entityClass = ScalingActivity.class)
 @Transactional(Transactional.TxType.REQUIRED)
-public interface ScalingActivityRepository extends CrudRepository<ScalingActivity, Long> {
+public interface ScalingActivityRepository extends PagingAndSortingRepository<ScalingActivity, Long> {
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.activityCrn = :activityCrn")
     Optional<ScalingActivity> findByActivityCrn(@Param("activityCrn") String activityCrn);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId")
-    List<ScalingActivity> findAllByCluster(@Param("clusterId") Long clusterId);
+    List<ScalingActivity> findAllByCluster(@Param("clusterId") Long clusterId, Pageable pageable);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime < :startTimeBefore ORDER BY st.startTime DESC")
     List<ScalingActivity> findAllByClusterWithStartTimeBefore(@Param("clusterId") Long clusterId, @Param("startTimeBefore") Date startTimeBefore);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime >= :startTimeAfter ORDER BY st.startTime DESC")
-    List<ScalingActivity> findAllByClusterWithStartTimeAfter(@Param("clusterId") Long clusterId, @Param("startTimeAfter") Date startTimeAfter);
+    List<ScalingActivity> findAllByClusterWithStartTimeAfter(@Param("clusterId") Long clusterId,
+            @Param("startTimeAfter") Date startTimeAfter, Pageable pageable);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.activityStatus = :activityStatus AND st.startTime >= :startTimeFrom " +
             "AND st.startTime < :startTimeUntil ORDER BY st.startTime DESC")
     List<ScalingActivity> findAllByClusterAndActivityStatusBetweenInterval(@Param("clusterId") Long clusterId,
             @Param("activityStatus") ActivityStatus activityStatus, @Param("startTimeFrom") Date startTimeFrom, @Param("startTimeUntil") Date startTimeUntil);
 
+    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.activityStatus IN :statuses AND" +
+            " st.startTime >= :startTimeAfter ORDER BY st.startTime DESC")
+    List<ScalingActivity> findAllByClusterAndInStatusesAndInGivenInterval(@Param("clusterId") Long clusterId,
+            @Param("statuses") Collection<ActivityStatus> statuses, @Param("startTimeAfter") Date startTimeAfter, Pageable pageable);
+
```

Comment:
The method name is incorrect here - It says "interval" but you're retrieving values `>=startTimeAfter`

Reply:
Done